File: src/sdk-reference/csharp/1/realtime/subscribe/snippets/message-notifications.cs
NotificationListener listener =
  [](const notification_result *notification) {
    Console.WriteLine("Message notification received");

    Console.WriteLine(notification.result.content);
    // { "metAt": "Insane", "hello": "world" }
  };

try {
  // Subscribes to notifications for documents
  kuzzle.realtime.subscribe("i-dont-exist", "in-database", "{}", listener);

  string message = @"{ ""metAt"": ""Insane"", ""hello"": ""world"" }";
  kuzzle.realtime.publish("i-dont-exist", "in-database", message);
} catch (KuzzleException e) {
  Console.Error.WriteLine(e.Message());
}

File: src/sdk-reference/cpp/1/realtime/subscribe/snippets/message-notifications.cpp
kuzzleio::NotificationListener listener =
  [](const kuzzleio::notification_result *notification) {
    std::cout << "Message notification received" << std::endl;

    std::cout << notification->result->content << std::endl;
    // { "metAt": "Insane", "hello": "world" }
  };

try {
  // Subscribes to notifications for documents
  kuzzle->realtime->subscribe("i-dont-exist", "in-database", "{}", &listener);

  std::string message = R"({ "metAt": "Insane", "hello": "world" })";
  kuzzle->realtime->publish("i-dont-exist", "in-database", message);
} catch (kuzzleio::KuzzleException &e) {
  std::cerr << e.what() << std::endl;
}
